### PR TITLE
feat(platform,docs): table preserved state

### LINKED ISSUE #9752: [Sourcing] Tree table: New ability to store and recover last table state after editing table data
Is this a bug, enhancement, or feature request?
--
Feature request

Briefly describe your proposal.
--
When we try to edit/update row data in table via inputs on another page/view (we use Angular router to a different page), the last table state does not get persisted but instead gets a total refresh.​

For example, when a new row is dynamically added and table gets refreshed, table does not scroll into that position where the newly added row resides. We'd like to recover to that last table page position after performing any edit to table data.

Whenever user performs any type of editing to the row data on a new page (navigated via Angular router), the table view should retain its last viewed page when user navigates back from that editing page view.

We need a 'scrollTo' strategy so that we can customize which table page the table could display, instead of always showing the 1st table page (with the very first 20 rows or so) upon completing editing.

Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
--
Current version.

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

closes #9752

## Description
- Added ability to set initial page for the platform table;
- Added ability to set initial `expanded` state of the row;
- Added ability to load pages before initial page (works in conjunction with `pageScrolling` input property);
- Added output event for pagination state changes;
- Added output event for table scrollable body scroll;
- Added output event for cases when new table rows are being rendered;
- Added example showing how to preserve the table state and data, and restore it when table is re-initialized;
- `fixed p13 filter dialog fields vertical alignment.

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Comment by g-cheishvili** [comment]
Something's wrong with the Preserved state example code loading

**Comment by N1XUS** [comment]
@g-cheishvili fixed issue with invalid example code link

**Comment by g-cheishvili** [comment]
e2e fails, probably some memory leak?

**Comment by N1XUS** [comment]
@g-cheishvili strange. The new example page shouldn't be tested at all, and the changes done to a table component should not generate any leakage

File: libs/docs/platform/table/examples/preserved-state/platform-table-preserved-state-example.component.ts [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+
+    columns: string[] = ['name', 'price', 'status'];
+
+    groupBy: CollectionGroup[] = [];
+
+    items = buildTree([...ITEMS]);
+
+    source = new ExampleTableDataSource(new ExampleTableProvider(this.items));
+
+    // source = of(ITEMS);
```
Comment (mikerodonnell89) [inline libs/docs/platform/table/examples/preserved-state/platform-table-preserved-state-example.component.ts:51]: Can remove this I think

File: libs/docs/platform/table/examples/preserved-state/platform-table-preserved-state-example.component.ts [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+} from '@fundamental-ngx/platform/table';
+import { map, Observable, of } from 'rxjs';
+
+@Component({
+    selector: 'fdp-platform-table-preserved-state-example',
+    templateUrl: './platform-table-preserved-state-example.component.html',
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PlatformTablePreservedStateExampleComponent {
```
Comment (mikerodonnell89) [inline libs/docs/platform/table/examples/preserved-state/platform-table-preserved-state-example.component.ts:37]: I understand the purpose of this example is to illustrate the ability to save options regarding the state of the table, but I think we will see some confusion over the sort/group/filter not actually working here. Let's copy over sort/group/filter algorithms from other examples so it actually works here too

**Pull request merged**